keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,233 @@
+
+We use this new configuration for all integrations in the same way that the current
+`namespaceSelector` is used by the Pod integration.  Specifically,
+1. If `manageJobsWithoutQueueName` is false, `managedJobsNamespaceSelector` has no effect: Kueue will manage
+exactly those instances of supported Kinds that have a `queue-name` label.
+2. If `manageJobsWithoutQueueName` is true, then Kueue will (a) manage all instances of supported Kinds
+that have a `queue-name` label and (b) will manage all instances of supported Kinds that do not
+have a `queue-name` label if they are in namespaces that match `managedJobsNamespaceSelector`.
+
+We migrate away from using `podOptions.namespaceSelector` over the course of several releases.
+Ideally we remove it from the `Configuration` struct as part of going to the `v1` API level.
+
+### User Stories (Optional)
+
+#### Story 1
+
+Cluster admins will be able to use the combination of `managedJobsNamespaceSelector` and
+`manageJobsWithoutQueueName` to configure Kueue to enable quota enforcement on a per namespace basis.
+
+One sapproach would be to exempt selected namespaces from management:
+```yaml
+managedJobsNamespaceSelector:
+  matchExpressions:
+  - key: kubernetes.io/metadata.name
+    operator: NotIn
+    values: [ kube-system, kueue-system ]
+```
+Another approach would be for the cluster admin to label namespaces that are subject to management:
+```yaml
+managedJobsNamespaceSelector:
+  matchLabels:
+    kueue.x-k8s.io/managed-namespace: true
+```
+
+## Design Details
+
+### API
+
+The `Configuration` struct is extended to add ManageAllJobsNamespaceSelector
+```go
+type Configuration struct {
+
+  ...
+
+	// ManageJobsWithoutQueueName controls whether or not Kueue reconciles
+	// jobs that don't set the label kueue.x-k8s.io/queue-name.
+	// If set to true, then those jobs will be suspended and never started unless
+	// they are assigned a queue and eventually admitted. This also applies to
+	// jobs created before starting the kueue controller.
+	// Defaults to false; therefore, those jobs are not managed and if they are created
+	// unsuspended, they will start immediately.
+	ManageJobsWithoutQueueName bool `json:"manageJobsWithoutQueueName"`
+
+  // ManageAllJobsNamespaceSelector can be used to omit some namespaces from ManageJobsWithoutQueueName
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+
+  ...
+}
+```
+
+### Implementation
+
+The configuration of all Job webhooks are extended to propagate the namespace selector and it is
+used to filter `manageJobsWithoutQueueName` as is currently done in the Pod integration. Specifically,
+jobs without queue names are only managed if both `manageJobsWithoutQueueName` is true and
+the jobs namespace matches the `managedJobsNamespaceSelector`.
+
+Configuration parsing during startup is extended to give a deprecation notice if `podOptions.namespaceSelector` is set.
```

Comment:
One note, the proposed configuration doesn't have a way of impersonating `podOptions.podSelector`. Would we be okay with deprecating it anyways?

Reply:
I think `podOptions.podSelector` is orthogonal to `podOptions.namespaceSelector`.  This KEP is only proposing to deprecate `podOptions.namespaceSelector` in favor of the new namespaceSelector that would apply to all integrations.